### PR TITLE
V5 download code review notes

### PR DESCRIPTION
**Pull request by rem-luminoso**

### CODE REVIEW

File: luminoso_api/v5_download.py [hunk 1]
```diff
@@ -32,14 +32,15 @@ def iterate_docs(client, expanded=False, progress=False):
...
+    its analysis, such as 'terms' and 'vector'.
 
     Otherwise, it will contain only the fields necessary to reconstruct the
     document: 'title', 'text', and 'metadata'.
 
     Shows a progress bar if progress=True.
     """
-    num_docs = client.get('docs', limit=1)['total_count']
+    # Get total number of docs from the project record
+    num_docs = client.get()['counts']['__all__']
```
Comment (Tahnan) [inline luminoso_api/v5_download.py]: `['counts']['__all__']` is a v4 API return format.  For v5, you want `['document_count']`.

File: luminoso_api/v5_download.py [hunk 2]
```diff
@@ -97,7 +96,8 @@ def main():
     Handle arguments for the 'lumi-download' command.
     """
     parser = argparse.ArgumentParser(
-        description=DESCRIPTION, formatter_class=argparse.RawDescriptionHelpFormatter
+        description=DESCRIPTION,
```
Comment (Tahnan) [inline luminoso_api/v5_download.py]: Nitpick: trailing whitespace.

File: luminoso_api/v5_download.py [hunk 3]
```diff
@@ -118,16 +118,20 @@ def main():
...
+        'output_file', nargs='?', default=None,
+        help='The JSON lines (.jsons) file to write to'
     )
     args = parser.parse_args()
 
     if args.save_token:
         if not args.token:
             raise ValueError('error: no token provided')
-        LuminosoClient.save_token(args.token, domain=urlparse(args.base_url).netloc)
+        LuminosoClient.save_token(args.token,
```
Comment (Tahnan) [inline luminoso_api/v5_download.py]: Nitpick: trailing whitespace.

**Comment by rem-luminoso** [comment]
I did look in the ProjectDB code to see if I could find what fields the project should have, but I somehow missed the list at the top of the file, which I see now.  (I got `'counts': {'__all__': X}` off a random project in master-staging.)  I'm still not sure which ones it would make sense to include here.  Do you have an opinion?

**Pull request merged**